src/main/scala/zio/s3/Live.scala

```diff
@@ -95,6 +95,10 @@ final class Live(unsafeClient: S3AsyncClient) extends S3.Service {
         ).map(S3ObjectListing.fromResponse)
       }
 
+  override def listAllObjects(bucketName: String, prefix: String): ZStream[Any, S3Exception, S3ObjectSummary] =
+    executePublisher(_.listObjectsV2Paginator(ListObjectsV2Request.builder().bucket(bucketName).prefix(prefix).build()))
```

Comment:
https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/s3/S3AsyncClient.html#listObjectsV2Paginator-software.amazon.awssdk.services.s3.model.ListObjectsV2Request-

`Returns some or all (up to 1,000) of the objects in a bucket. `

Reply:
The documentation has a not in the bottom 
```
Please notice that the configuration of MaxKeys won't limit the number of results you get with the paginator. It only limits the number of results in each page.
```
I added a test in order to show the difference.
So essentially the `*Paginator` methods in the aws sdk are implemented the same way the `listObjectsDescendant` is implemented in zio-s3: they traverse over all the pages and return them a publisher.

Note 1: while implementing the test I found an issue with the Test s3 implementation. The files that are named like `1001obj/0YM2XhBT5o` so containing `/` in the name cannot be uploaded because the intermediate folders are not created on the disk. I added the operation to create intermediate folders. The querying by prefix also works.

Note 2: I changed the tests to run sequentially. `putObject` creates files and the parallel tests that list files can see this uploaded objects. This behaviour wasn't visible on a couple of uploads, but it is occurring with a lot of files (1000 in the new test).
An alternative solution would be to have a subfolder per test so that the upload/listing operations in parallel tests don't interfere with each other.